pkg/utils/generate.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// PodArg by default use `Status=corev1.PodRunning` and `Namespace=metav1.NamespaceDefault`.
+// For the others, the default values are empty.
+type PodArg struct {
+	Name            string
+	Status          v1.PodPhase `default:"corev1.PodRunning"`
+	Namespace       string      `default:"metav1.NamespaceDefault"`
```

Comment:
"struct tag" always be used for describing specific logic on a field, like `json`, `protobuf`

 It seems we do not import any library for that, so I think we should delete `default` struct tag.

Reply:
Thanks! I didn't know that before.

Actually I got the idea from [go-defaults](https://github.com/mcuadros/go-defaults). Then I guess I would either use that library or explicitly use `reflect` to get the tag value. Which one do you think is better

Reply:
It seems you need to call `defaults.SetDefaults()` for setting default values, we often do it in the constructor. So you might need a constructor. 🙈

So how about deleting the "default" struct tag?

Reply:
As the [caveats of repo](https://github.com/mcuadros/go-defaults#caveats) shows
> At the moment, the way the default filler checks whether it should fill a struct field or not is by comparing the current field value with the corresponding zero value of that type.

So we could still put `defaults.SetDefaults()` into `newPod`, replacing the manual default values set.

I think the tags could somewhat highlight the default values, but I'm still ok to delete them.

Reply:
> So we could still put `defaults.SetDefaults()` into `newPod`, replacing the manual default values set.

Please don't, it makes `newPod` more and more complicated, just remove it. 🙈

Reply:
Got it. Deleted